### ISSUE TITLE
Bug in SSH Connection in PyScripter: Error Copying/Reading Files (Proposed Fix Included)

### ISSUE THREAD

**Issue by ebalvis** [pyscripter/pyscripter#1351]
Dear PyScripter Development Team,

I would like to report a bug related to the SSH connection functionality in PyScripter (Run > Python Engine > SSH). When using this option, I encountered errors while copying or reading files during script execution.

After investigating the issue, I found that the error seems to be related to the handling of single (') and double (") quotes in file paths when establishing a remote connection.

I implemented a change in the code that resolved the problem in my environment. Below is the modification I applied:

```
function TSSHServices.Scp(const ScpCommand, FromFile, ToFile: string; out ErrorMsg: string; ScpOptions: string): boolean;

  if FromFile.Contains('@') then
    aFromFile := FromFile.Replace('''', '').Replace('"', '')
  else
    aFromFile := FromFile;
  if ToFile.Contains('@') then
    aToFile := ToFile.Replace('''', '').Replace('"', '')
  else
    aToFile := ToFile;
  Command := Format('"%s" %s %s %s', [ScpCommand, ScpOptions, aFromFile, aToFile]);
```

This adjustment removes single and double quotes from the strings associated with file paths in remote connections. With this fix, the error stopped occurring, and file copy and read operations worked as expected.

I am happy to provide additional details or assist with further testing if needed.

Thank you for your attention and for the excellent work on PyScripter.

Best regards,

**Comment by pyscripter** [comment]
Is this related to https://pyscripter.blogspot.com/2024/11/ssh-engines-broken-by-november-12-2024.html?

Does the fix suggested there solves the issue?

**Comment by ebalvis** [comment]
The issue seems to be related to the quotation marks used in the SCP command generated by PyScripter.

The failing command:


```
"scp" -T -o PasswordAuthentication=no -o StrictHostKeyChecking=no "C:\Users\ebalvis.DESKTOP-67C66EV.000\AppData\Roaming\PyScripter\remserver.py" "root@192.168.88.56:''/tmp/remserver.py''"
has additional quotes around the destination '/tmp/remserver.py', which causes the error.
```

The working command:


```
"scp" -T -o PasswordAuthentication=no -o StrictHostKeyChecking=no "C:\Users\ebalvis.DESKTOP-67C66EV.000\AppData\Roaming\PyScripter\remserver.py" root@192.168.88.56:/tmp/remserver.py
does not include those extra quotes around the destination.
```

It seems PyScripter is generating the SCP command incorrectly. Perhaps an update or fix to the part of the script responsible for constructing the command could resolve this.

Additionally, I tried the fix suggested in https://pyscripter.blogspot.com/2024/11/ssh-engines-broken-by-november-12-2024.html, but the same error persists.

I hope this information helps diagnose the issue.

**Comment by pyscripter** [comment]
What is the version of your scp? (ssh -V)

```
PS C:\> ssh -V
OpenSSH_for_Windows_9.5p1, LibreSSL 3.8.2
```

See https://www.openssh.com/releasenotes.html.

> Changes since OpenSSH 8.9
> =========================
> 
> This release is focused on bug fixing.
> 
> Potentially-incompatible changes
> --------------------------------
> 
> This release switches [scp(1)](https://man.openbsd.org/scp.1) from using the legacy scp/rcp protocol
> to using the SFTP protocol by default.
> 
> Legacy scp/rcp performs wildcard expansion of remote filenames (e.g.
> "scp host:* .") through the remote shell. This has the side effect of
> requiring double quoting of shell meta-characters in file names
> included on [scp(1)](https://man.openbsd.org/scp.1) command-lines, otherwise they could be interpreted
> as shell commands on the remote side.
> 
> This creates one area of potential incompatibility: [scp(1)](https://man.openbsd.org/scp.1) when using
> the SFTP protocol no longer requires this finicky and brittle quoting,
> and attempts to use it may cause transfers to fail. We consider the
> removal of the need for double-quoting shell characters in file names
> to be a benefit and do not intend to introduce bug-compatibility for
> legacy scp/rcp in [scp(1)](https://man.openbsd.org/scp.1) when using the SFTP protocol.
> 
> Another area of potential incompatibility relates to the use of remote
> paths relative to other user's home directories, for example -
> "scp host:~user/file /tmp". The SFTP protocol has no native way to
> expand a ~user path. However, [sftp-server(8)](https://man.openbsd.org/sftp-server.8) in OpenSSH 8.7 and later
> support a protocol extension "expand-path@openssh.com" to support
> this.
> 
> In case of incompatibility, the [scp(1)](https://man.openbsd.org/scp.1) client may be instructed to use
> the legacy scp/rcp using the -O flag.

Before the November Windows update double quoting was required.   After the update double quoting results in errors unless  the -O flag is specified.

I cannot change the quoting code because it will break ssh with earlier versions of openssh.    But here adding the - O flag solves the issue.

**Comment by pyscripter** [comment]
This works fine here:

![image](https://github.com/user-attachments/assets/8ffcfa13-e1bb-4309-b212-55529238ab8c)
![image](https://github.com/user-attachments/assets/de8149f2-2f00-4704-8f38-97668e178df3)

**Comment by ebalvis** [comment]
After applying the suggested solution (adding the -O option in the SSH configuration), the connection now works perfectly without errors. However, I have encountered a new issue while debugging applications.

When running an application in debug mode, the code does not stop at the breakpoints I set. Instead, the program runs completely as if no breakpoints were defined, ignoring them entirely.

Environment Details:

PyScripter Version: 5.1.2.0 x64
SSH Connection: Configured with the -O option and working properly
The issue occurs even with simple code that should pause at breakpoints
No error messages or warnings appear in the debug console
![image](https://github.com/user-attachments/assets/eda88c0a-06f3-4a1b-92e7-a84b43a4c58c)
